arrays/arrays.c

```diff
@@ -83,15 +98,27 @@ char *arr_read(Array *arr, int index) {
 void arr_insert(Array *arr, char *element, int index) {
 
   // Throw an error if the index is greater than the current count
+  if (index > arr->count) {
+    fprintf(stderr, "Index is greater than the current count\n");
+    exit(1);
+  }
 
   // Resize the array if the number of elements is over capacity
+  if (arr->count + 1 > arr->capacity) {
+    resize_array(arr);
+  }
 
   // Move every element after the insert index to the right one position
+  for (int i = index; i < arr->count; i++) {
```

Comment:
You should use our for loop to traverse the array from right to left. If you do it this way you're probably going to copy the last value twice and it won't function the way you want. You should experiment with it though to see what I mean.